src/main/java/org/fogbowcloud/saps/engine/core/scheduler/Scheduler.java

```diff
@@ -70,126 +56,27 @@ public Scheduler(Properties properties, Catalog catalog, ScheduledExecutorServic
-            LOGGER.error("Required property " + SapsPropertiesConstants.IMAGE_DATASTORE_PORT + " was not set");
-            return false;
-        }
-        if (!properties.containsKey(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_SUBMISSOR)) {
-            LOGGER.error(
-                    "Required property " + SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_SUBMISSOR + " was not set");
-            return false;
-        }
-        if (!properties.containsKey(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_CHECKER)) {
-            LOGGER.error("Required property " + SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_CHECKER + " was not set");
-            return false;
-        }
-        if (!properties.containsKey(SapsPropertiesConstants.ARREBOL_BASE_URL)) {
-            LOGGER.error("Required property " + SapsPropertiesConstants.ARREBOL_BASE_URL + " was not set");
-            return false;
-        }
+        String[] propertiesSet = {
+                SapsPropertiesConstants.IMAGE_DATASTORE_IP,
+                SapsPropertiesConstants.IMAGE_DATASTORE_PORT,
+                SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_SUBMISSOR,
+                SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_CHECKER,
+                SapsPropertiesConstants.ARREBOL_BASE_URL
 
-        LOGGER.debug("All properties are set");
-        return true;
+        };
+
+        return SapsPropertiesUtil.checkProperties(properties, propertiesSet);
     }
 
-    /**
-     * Start Scheduler component
-     *
-     * @throws Exception
-     */
-    public void start() throws Exception {
+    public void start() {
         recovery();
 
-        try {
-            sapsExecutor.scheduleWithFixedDelay(new Runnable() {
-                                                    @Override
-                                                    public void run() {
-                                                        schedule();
-                                                    }
-                                                }, 0, Integer.parseInt(properties.getProperty(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_SUBMISSOR)),
-                    TimeUnit.SECONDS);
-
-            sapsExecutor.scheduleWithFixedDelay(new Runnable() {
-                                                    @Override
-                                                    public void run() {
-                                                        checker();
-                                                    }
-                                                }, 0, Integer.parseInt(properties.getProperty(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_CHECKER)),
-                    TimeUnit.SECONDS);
-
-        } catch (Exception e) {
-            LOGGER.error("Error while starting Scheduler component", e);
-        }
+        sapsExecutor.scheduleWithFixedDelay(() -> schedule(), 0, Integer.parseInt(properties.getProperty(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_SUBMISSOR)),
```

Comment:
using properties deep in this code is bad

Reply:
Added fixme.